src/tests/Microsoft.PowerFx.Core.Tests/IntellisenseTests/TokenizationTests/ExpectedToken.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Texl.Intellisense;
+
+namespace Microsoft.PowerFx.Tests.IntellisenseTests
+{
+    internal class ExpectedToken : ITokenTextSpan
+    {
+        /// <summary>
+        /// Special start index to indicate that start index shoud be computed based on previous token end index.
+        /// </summary>
+        public const int SpecialStartIdx = -1;
+
+        public int StartIndex { get; set; } = SpecialStartIdx;
+
+        public int Length { get; set; }
+
+        public TokenType TokenType { get; set; }
+
+        public int EndIndex => Length + StartIndex;
+
+        public bool CanBeHidden { get; set; } = false;
```

Comment:
>set [](http://example.com/codeflow?start=39&length=3)

can these be readonly?  easier to verify in threading cases.

Reply:
Yeah. I could convert them to readonly. I need setter for StartIndex but i would instead create a new instance of expected token and make start index readonly.